src/main/java/org/icgc_argo/workflow/search/service/wes/WesRunService.java

```diff
@@ -18,260 +18,128 @@
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import static java.lang.String.format;
+import static org.icgc_argo.workflow.search.model.EsDefaults.ES_PAGE_DEFAULT_SIZE;
+import static org.icgc_argo.workflow.search.model.wes.State.fromValue;
+import static org.icgc_argo.workflow.search.util.Converter.buildRunLog;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 import lombok.NonNull;
-import lombok.Value;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import lombok.val;
-import org.elasticsearch.action.search.SearchRequest;
-import org.elasticsearch.action.search.SearchResponse;
-import org.elasticsearch.client.RequestOptions;
-import org.elasticsearch.client.RestHighLevelClient;
-import org.elasticsearch.index.query.QueryBuilders;
-import org.elasticsearch.search.SearchHit;
-import org.elasticsearch.search.aggregations.AggregationBuilder;
-import org.elasticsearch.search.aggregations.AggregationBuilders;
-import org.elasticsearch.search.aggregations.bucket.MultiBucketsAggregation;
-import org.elasticsearch.search.aggregations.bucket.terms.Terms;
-import org.elasticsearch.search.builder.SearchSourceBuilder;
-import org.elasticsearch.search.sort.SortOrder;
-import org.icgc_argo.workflow.search.config.ElasticsearchProperties;
 import org.icgc_argo.workflow.search.config.ServiceInfoProperties;
-import org.icgc_argo.workflow.search.index.model.TaskDocument;
-import org.icgc_argo.workflow.search.index.model.WorkflowDocument;
+import org.icgc_argo.workflow.search.model.common.Task;
 import org.icgc_argo.workflow.search.model.exceptions.NotFoundException;
 import org.icgc_argo.workflow.search.model.wes.*;
-import org.icgc_argo.workflow.search.service.annotations.HasQueryAccess;
+import org.icgc_argo.workflow.search.service.graphql.RunService;
+import org.icgc_argo.workflow.search.service.graphql.TaskService;
 import org.icgc_argo.workflow.search.util.Converter;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import static java.lang.String.format;
-import static org.icgc_argo.workflow.search.model.SearchFields.*;
-import static org.icgc_argo.workflow.search.util.Converter.buildRunLog;
-import static org.icgc_argo.workflow.search.util.Converter.convertSourceMapToRunStatus;
+import reactor.core.publisher.Mono;
 
+/**
+ * The WesRunService is just a shim over the graphql services, used to convert the responses from
+ * them to wes standard objects. All logic regarding repository query, security etc. is handled in
+ * those services.
+ */
```

Comment:
I removed all the es client logic as well since we already have it in the Run+Task Service/Repo.